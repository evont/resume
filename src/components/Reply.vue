<template>
  <div class="m-reply">
    <div class="u-dialog" data-isquestion="1" is="transition-group" name="reply-dialog">
      <div class="detail"
        v-for="(item, index) in replies"
        :key="index"
        @click="$emit('reply', item);"
        v-html="htmlContent(item)">
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Reply',
  props: ['replies'],
  methods: {
    htmlContent(content) {
      return `<${content.tag}>${content.detail}</${content.tag}>`;
    },
  },
};
</script>
