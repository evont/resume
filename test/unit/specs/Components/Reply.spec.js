import { mount } from 'vue-test-utils'
import Reply from '@/components/Reply';

const replies = [
  {
    tag: 'p',
    detail: 'abacadabara'
  }
]
describe('Reply.vue', () => {
  let cmp
  beforeEach(() => {
    cmp = mount(Reply, {
      sync: true,
      propsData: {
        replies
      }
    })
  })
  it('should emitReply', () => {
    cmp.find('.detail').trigger('click') // 对button触发一个click事件
    expect(cmp.emitted('reply')).toBeTruthy() // 查看render2Pic是否被调用
  });

  it('should has element', () => {
    expect(cmp.findAll('.detail').length).toBe(replies.length)
  });

  it('should reply something', () => {
    const content = replies[0];
    expect(cmp.find('.detail').html()).toBe(`<div class=\"detail\"><${content.tag}>${content.detail}</${content.tag}></div>`)
  });

  it('Should have the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot() // 调用toMatchSnapshot来比对快照
  })
});
  