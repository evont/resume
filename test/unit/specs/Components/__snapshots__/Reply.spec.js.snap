// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reply.vue Should have the expected html structure 1`] = `
<div
  class="m-reply"
>
  <div
    class="u-dialog"
    data-isquestion="1"
    name="reply-dialog"
  >
    <div
      class="detail"
    >
      <p>
        abacadabara
      </p>
    </div>
  </div>
</div>
`;
